storage/badger/my_receipts.go

```diff
@@ -0,0 +1,121 @@
+package badger
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/dgraph-io/badger/v2"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/storage/badger/operation"
+)
+
+// MyExecutionReceipts holds and indexes Execution Receipts.
+// MyExecutionReceipts is implemented as a wrapper around badger.ExecutionReceipts
+// The wrapper ads the ability to "MY execution receipt", from the viewpoint
+// of an individual Execution Node.
+type MyExecutionReceipts struct {
+	genericReceipts *ExecutionReceipts
+	db              *badger.DB
+	cache           *Cache
+}
+
+func NewMyExecutionReceipts(collector module.CacheMetrics, db *badger.DB, receipts *ExecutionReceipts) *MyExecutionReceipts {
+	store := func(key interface{}, val interface{}) func(tx *badger.Txn) error {
+		receipt := val.(*flow.ExecutionReceipt)
+		// assemble DB operations to store receipt (no execution)
+		storeReceiptOps := receipts.store(receipt)
+		// assemble DB operations to index receipt as one of my own (no execution)
+		blockID := receipt.ExecutionResult.BlockID
+		receiptID := receipt.ID()
+		indexOwnReceiptOps := func(tx *badger.Txn) error {
+			err := operation.IndexOwnExecutionReceipt(blockID, receiptID)(tx)
+			// check if we are storing same receipt
+			if errors.Is(err, storage.ErrAlreadyExists) {
+				var savedReceiptID flow.Identifier
+				err := operation.LookupOwnExecutionReceipt(blockID, &savedReceiptID)(tx)
+				if err != nil {
+					return err
+				}
+
+				// if we are storing same receipt we shouldn't error
+				if savedReceiptID == receiptID {
```

Comment:
If already exists but doesn't match with the saved receipt ID, I would suggest to include the saved receipt ID in the the error to return , helpful for debugging.